transforms/builder/addAttribute.js

```diff
@@ -0,0 +1,57 @@
+const addAttribute = (path, attribute, value) => (
+  fileInfo,
+  api,
+  options,
+  j,
+  root
+) => {
+  const components = root.findJSXElementsByImport(path);
+
+  if (components.size() === 0) {
+    // If the component is not imported, skip this file
+    return;
+  }
+  if (!components) {
+    return false;
+  }
+
+  let didUpdate = false;
+  components.forEach((component) => {
+    if (
+      !j(component.value.openingElement.attributes)
+        .find(j.JSXIdentifier, { name: attribute })
+        .size()
+    ) {
+      if (typeof value === "string") {
+        component.value.openingElement.attributes.push(
+          j.jsxAttribute(j.jsxIdentifier(attribute), j.literal(value))
+        );
+        didUpdate = true;
```

Comment:
A performance boost would just be to return true here rather than setting `didUpdate` to true and still running over the rest of the if statements. No point running over those if we've already added the new prop here. 

It would also mean you wouldn't need to check for `typeof value !== "string"` below because we'd know if it got to that point it wasn't a string

Same goes for the other if statements below